bokehjs/src/lib/models/tools/actions/fullscreen_tool.ts

```diff
@@ -0,0 +1,37 @@
+import {logger} from "core/logging"
+import * as p from "core/properties"
+import {ActionTool, ActionToolView} from "models/tools/actions/action_tool"
+import screenfull from "screenfull"
+
+export class FullscreenToolView extends ActionToolView {
+  model: FullscreenTool
+
+  doit(): void {
+    if (screenfull.isEnabled)
+      screenfull.toggle((this.root as any).el);
```

Comment:
Presumably we should make it configurable which element it actually fullscreens, i.e. either the plot(s) the toolbar is attached to or the root. It seems that the `el` is not defined on the interface, hence I've used the `any` for now.

Reply:
`root` is defined on `View`, but `el` comes with `DOMView`. Given that currently on DOM views can be roots (or in fact layoutable DOM views), this is wrong and results in this cast.

Reply:
It's a bit awkward to have this tool act on the root, but I suppose there isn't any way otherwise to give users access to a tool. Having a single tool toolbar would be an overkill. Maybe a context menu could be used (I introduced them recently on the canvas in the grid plot work).

A bigger issue is that this isn't going to work with proxy toolbar, if you have an even number of full screen tools proxied (every other tool will cancel out). Tough, given that fullscreen API is asynchronous, maybe it will work by chance, i.e. all requests will accumulate before the first promise is resolved.

Reply:
For grid plots I'm working on "global tools" (e.g. save tool) and in general I would like to get away from naively proxying tools, so maybe this will be a non-issue at some point.